test/udp_sender.js

```diff
@@ -26,251 +26,269 @@ import ThriftUtils from '../src/thrift.js';
+          done();
+        },
+      };
+      let span = tracer.startSpan(undefined);
+      span.finish();
+
+      sender.append(ThriftUtils.spanToThrift(span), (numSpans, err) => {
+        assert.equal(numSpans, 1);
+        expect(err).to.have.string('error converting span to Thrift:');
         done();
-      },
-    };
-    let span = tracer.startSpan(undefined);
-    span.finish();
-
-    sender.append(ThriftUtils.spanToThrift(span), (numSpans, err) => {
-      assert.equal(numSpans, 1);
-      expect(err).to.have.string('error converting span to Thrift:');
-      done();
+      });
     });
-  });
 
-  it('should return error on span too large', done => {
-    let span = tracer.startSpan('op-name');
-    span.finish(); // otherwise duration will be undefined
+    it('should return error on span too large', done => {
+      let span = tracer.startSpan('op-name');
+      span.finish(); // otherwise duration will be undefined
 
-    sender._maxSpanBytes = 1;
-    sender.append(ThriftUtils.spanToThrift(span), (numSpans, err) => {
-      assert.equal(numSpans, 1);
-      expect(err).to.have.string('is larger than maxSpanSize');
-      done();
+      sender._maxSpanBytes = 1;
+      sender.append(ThriftUtils.spanToThrift(span), (numSpans, err) => {
+        assert.equal(numSpans, 1);
+        expect(err).to.have.string('is larger than maxSpanSize');
+        done();
+      });
     });
-  });
-
-  it('should return 0,undefined on flush() with no spans', () => {
-    sender.flush(assertCallback(0, undefined));
-  });
 
-  it('should gracefully handle errors emitted by socket.send', function(done) {
-    // this test tends to timeout
-    this.timeout(15000);
-    let tracer = new Tracer('test-service-name', new RemoteReporter(sender), new ConstSampler(true));
-    sender._host = 'foo.bar.xyz';
-    // In Node 0.10 and 0.12 the error is logged twice: (1) from inline callback, (2) from on('error') handler.
-    let expectLogs = semver.satisfies(process.version, '0.10.x || 0.12.x');
-    sender._logger = {
-      info: msg => {
-        console.log('sender info: ' + msg);
-      },
-      error: msg => {
-        assert.isTrue(expectLogs);
-        expect(msg).to.have.string('error sending spans over UDP: Error: getaddrinfo ENOTFOUND');
-        tracer.close(done);
-      },
-    };
-    tracer.startSpan('testSpan').finish();
-    sender.flush((numSpans, err) => {
-      assert.equal(numSpans, 1);
-      expect(err).to.have.string('error sending spans over UDP: Error: getaddrinfo ENOTFOUND');
-      if (!expectLogs) {
-        tracer.close(done);
-      }
+    it('should return 0,undefined on flush() with no spans', () => {
+      sender.flush(assertCallback(0, undefined));
     });
-  }).timeout(5000);
-});
+
+    it('should gracefully handle errors emitted by socket.send', function(done) {
+      // EAI_AGAIN - nodejs received invalid DNS response. E.g. resolver doesn't support IPv6.
```

Comment:
When would this error happen in the test?

Reply:
This errors occurs when dns resolver doesn't support IPv6 or host doesn't have ipv6 resolver. I've got this error in travis for upd6 and `foo.bar.xyz` instead of `ENOTFOUND`.

Reply:
For example,
on my local Mac I've got `ENOTFOUND` for this domain because I have dns resolver.
In travis I've got `EAI_AGAIN`